code/game/gamemodes/dynamic/dynamic.dm

```diff
@@ -841,7 +841,7 @@ GLOBAL_LIST_EMPTY(dynamic_station_traits)
  * rand() calls without arguments returns a value between 0 and 1, allowing for smaller intervals.
  */
 /datum/game_mode/dynamic/proc/lorentz_to_amount(centre = 0, scale = 1.8, max_threat = 100, interval = 1)
-	var/location = rand(-MAXIMUM_DYN_DISTANCE, MAXIMUM_DYN_DISTANCE) * rand()
+	var/location = RANDOM_DECIMAL(-MAXIMUM_DYN_DISTANCE, MAXIMUM_DYN_DISTANCE) * rand()
```

Comment:
```suggestion
	var/location = RANDOM_DECIMAL(-MAXIMUM_DYN_DISTANCE, MAXIMUM_DYN_DISTANCE)
```
I don't think the extra rand() is necessary at this point

Reply:
That extra `rand()` creates bias towards the middle of the range, effectively creating a peak.

This is because random decimals closer to 0 are barely changed at all whether that rand() is closer to 1 or to 0.
Random decimals further from 0 can be changed significantly.

In order to get a threat value in the middle of the table, you can either roll a low number on your RANDOM_DECIMAL or a low rand() multiplier, or both.

In order to get a threat value at an extreme - like 0 threat or 100 threat - you need to BOTH roll a RANDOM_DECIMAL of -5 or 5 AND roll a rand() close to 1.

If I'm correct in my analysis of the code and writing the dynamic emulator last night, this rand() is the only thing making our threat distribution non-uniform.

Visualised in my Dynamic python port which makes different graphs shapes to Mothblocks probably due to how I'm representing the data, we get modified code with the rand():
![image](https://github.com/tgstation/tgstation/assets/24975989/53eb7adf-93c4-4cdb-a2c8-b1c7e973807a)

And without it:
![image](https://github.com/tgstation/tgstation/assets/24975989/07f5e1d8-517f-49ae-b2af-2c4c05cc9108)

Those 6 characters in `rand()` are doing **all** the work in generating any non-equal distribution of threat.

Reply:
Ooooooh, and I thought it was just an oversight.

Reply:
Oh, I was busy editing my previous comment while you were making further graphs.

Anyway, I haven't bothered asking why, but your 'standard settings' graph and mothblocks' look different. Theirs look like the sum of two binomial curves while yours look like one. both are good tho.

Reply:
Mothblocks' graph has a sample size an order of magnitude smaller than Timberpoes'. That could explain the discrepancy.

Reply:
My graph is 15,000 samples =P

Reply:
This is all good knowlewdge that hsould be in the repo as comments on this damn code